source/CIS/cis-0.rst

```diff
@@ -0,0 +1,117 @@
+.. _CIS-0:
+
+=========================
+CIS-0: Standard Detection
+=========================
+
+.. list-table::
+   :stub-columns: 1
+
+   * - Created
+     - Jul 14, 2022
+   * - Draft version
+     - 1
+   * - Supported versions
+     - | Smart contract version 1 or newer
+       | (Protocol version 4 or newer)
+   * - Standard identifier
+     - ``CIS-0``
+
+.. warning::
+
+   This standard is still a draft and significant changes might still be made.
+
+Abstract
+========
+
+A standardized interface for how a smart contract publish supported smart contract standards.
+Additionally, this specification includes a list of identifiers for each standard used on the Concordium blockchain.
+
+The interface provides an entrypoint for querying the smart contract, whether it supports a given standard. The smart contract respond with either: not supported, supported or supported by some contract address.
+
+Specification
+=============
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in :rfc:`2119`.
+
+.. _CIS-0-standard-identifer:
+
+Standard identifer
+-------------------
+
+Each standard MUST have a unique identifier.
+An identifier is a variable-length ASCII string up to 100 characters.
+
+The identifier is serialized as: first 1 byte for the length followed this many bytes for the ASCII encoding of the identifier::
+
+  StandardIdentifier ::= (n: Byte) (id: Byteⁿ)
+
+
+The table below contains the current standards and associated standard identifier.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 2 1 1
+
+   * - Standard
+     - Identifier
+     - Encoded
+   * - :ref:`CIS-0`
+     - ``CIS-0``
+     - ``054349532d30``
+   * - :ref:`CIS-1`
+     - ``CIS-1``
+     - ``054349532d31``
+   * - :ref:`CIS-2`
+     - ``CIS-2``
+     - ``054349532d32``
```

Comment:
I am not sure this table should be part of the standard since it means we have to keep updating the standard.

We need to have a better way of doing this.

Reply:
I have already added the identifier in the specification of each of the current standards.
So I'll should just remove this table.